src/dependencies.scala

```diff
@@ -150,5 +173,41 @@ object Dependencies {
         deps map { x => x: ClasspathDep[ProjectReference] }:_*)
     }
   }
+
+  implicit class ModuleIDOps(id: ModuleID)
+  {
+    def revCovers(other: String) = {
+      def partCovers(p: (String, String)) = p._1 == p._2 || p._1 == "+"
+      val parts = id.revision.toString.split('.')
+      val otherParts = other.toString.split('.')
+      val partsCover = parts zip(otherParts) forall(partCovers)
+      partsCover && (
+        parts.length == otherParts.length ||
+        (parts.length < otherParts.length) && parts.lastOption.exists(_ == "+")
+      )
+    }
+
+    def covers(other: ModuleID) = {
+      id.organization == other.organization && id.name == other.name &&
+        revCovers(other.revision)
+    }
```

Comment:
I don't care too much for the revCovers and covers naming. something like "revMatches" and "matches" would be preferable